File: VB/CustomEdit/Form1.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms

Namespace CustomEdit
	Partial Public Class Form1
		Inherits Form
		Private riCustomEdit As RepositoryItemCustomCheckedComboBoxEdit
		Public Sub New()
			InitializeComponent()

			Dim source As New List(Of GridRowData)()
			source.Add(New GridRowData())
			source.Add(New GridRowData())

			CreateRepository()

			'imageComboBoxEdit1
			FillImageComboBox()
			AddHandler imageComboBoxEdit1.EditValueChanged, AddressOf imageComboBoxEdit1_EditValueChanged

			'gridControl1
			gridControl1.DataSource = source
			gridControl1.RepositoryItems.Add(riCustomEdit)

			'gridView1
			gridView1.Columns("List").ColumnEdit = riCustomEdit
		End Sub
		Protected Overrides Sub Finalize()
			RemoveHandler imageComboBoxEdit1.EditValueChanged, AddressOf imageComboBoxEdit1_EditValueChanged
		End Sub

		Private Sub imageComboBoxEdit1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim edit As ImageComboBoxEdit = TryCast(sender, ImageComboBoxEdit)
			riCustomEdit.CheckStyle = CType(edit.EditValue, CheckStyles)
		End Sub

		Private Sub FillImageComboBox()
			imageComboBoxEdit1.Properties.Items.AddEnum(GetType(CheckStyles))
		End Sub

		Private Sub CreateRepository()
			riCustomEdit = New RepositoryItemCustomCheckedComboBoxEdit()
			riCustomEdit.Items.Add("Item 1")
			riCustomEdit.Items.Add("Item 2")
			riCustomEdit.Items.Add("Item 3")
		End Sub
	End Class
End Namespace

File: CS/CustomEdit/Form1.cs
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using System;
using System.Collections.Generic;
using System.Windows.Forms;

namespace CustomEdit
{
    public partial class Form1 : Form
    {
        RepositoryItemCustomCheckedComboBoxEdit riCustomEdit;
        public Form1()
        {
            InitializeComponent();

            List<GridRowData> source = new List<GridRowData>();
            source.Add(new GridRowData());
            source.Add(new GridRowData());

            CreateRepository();
            
            //imageComboBoxEdit1
            FillImageComboBox();
            imageComboBoxEdit1.EditValueChanged += imageComboBoxEdit1_EditValueChanged;

            //gridControl1
            gridControl1.DataSource = source;
            gridControl1.RepositoryItems.Add(riCustomEdit);

            //gridView1
            gridView1.Columns["List"].ColumnEdit = riCustomEdit;
        }
        ~Form1()
        {
            imageComboBoxEdit1.EditValueChanged -= imageComboBoxEdit1_EditValueChanged;
        }

        void imageComboBoxEdit1_EditValueChanged(object sender, EventArgs e)
        {
            ImageComboBoxEdit edit = sender as ImageComboBoxEdit;
            riCustomEdit.CheckStyle = (CheckStyles)edit.EditValue;
        }

        private void FillImageComboBox()
        {
            imageComboBoxEdit1.Properties.Items.AddEnum(typeof(CheckStyles));
        }

        private void CreateRepository()
        {
            riCustomEdit = new RepositoryItemCustomCheckedComboBoxEdit();
            riCustomEdit.Items.Add("Item 1");
            riCustomEdit.Items.Add("Item 2");
            riCustomEdit.Items.Add("Item 3");
        }
    }
}
